tools/analyzer_plugin/test/unit/analysis_options/parse_test.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:over_react_analyzer_plugin/src/analysis_options/parse.dart';
+import 'package:over_react_analyzer_plugin/src/analysis_options/plugin_analysis_options.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('PluginAnalysisOptions parse yaml', () {
+    test('with base case', () {
+      final options = processAnalysisOptionsFile(baseCaseYaml);
+      expect(options, isNotNull);
+      options!;
+      expect(options.errors, hasLength(4));
+      expect(options.errors['over_react_boilerplate_error'], equals(AnalysisOptionsSeverity.error));
+      expect(options.errors['over_react_incorrect_doc_comment_location'], equals(AnalysisOptionsSeverity.warning));
+      expect(options.errors['over_react_unnecessary_key'], equals(AnalysisOptionsSeverity.info));
+      expect(options.errors['over_react_pseudo_static_lifecycle'], equals(AnalysisOptionsSeverity.ignore));
+    });
+
+    test('without not configured', () {
+      final options = processAnalysisOptionsFile(pluginNotConfiguredYaml);
+      expect(options, isNull);
+    });
+
+    test('with no error section', () {
+      final options = processAnalysisOptionsFile(noErrorSectionYaml);
+      expect(options, isNotNull);
+      options!;
+      expect(options.errors, hasLength(0));
+    });
+
+    test('with invalid severity level', () {
+      final options = processAnalysisOptionsFile(invalidSeverityLevelYaml);
+      expect(options, isNotNull);
+      options!;
+      expect(options.errors, hasLength(2));
+      expect(options.errors['over_react_boilerplate_error'], equals(AnalysisOptionsSeverity.error));
+      expect(options.errors['over_react_pseudo_static_lifecycle'], equals(AnalysisOptionsSeverity.ignore));
```

Comment:
The `over_react_incorrect_doc_comment_location` code seems to have an invalid level, but I don't see a corresponding expect for it.

Can we assert in this test what the behavior is when the severity level is invalid?

Reply:
Done